torch/csrc/autograd/python_function.cpp

```diff
@@ -381,90 +378,104 @@ static void _transplant_var(VariableImpl& var, const std::shared_ptr<Function>&
 // do in this case.  After this method is run, t2var is extended with
 // mappings for output tensors as well.
 static void _wrap_outputs(THPFunction *self, t2var_type &t2var,
-    std::unordered_set<PyObject *> &dirty_inputs, PyObject *raw_output,
-    PyObject *outputs, bool is_volatile)
+    std::unordered_set<PyObject *> &dirty_inputs,
+    const t2var_type &shared_pairs,
+    PyObject *raw_output, PyObject *outputs, bool is_volatile)
 {
   auto cdata = is_volatile ? nullptr : THPFunction_asFunction(self);
   Py_ssize_t num_outputs = PyTuple_GET_SIZE(raw_output);
   if (self->cdata.is_executable) {
     self->output_info = new std::vector<output_info_type>();
     self->output_info->reserve(num_outputs);
   }
+
+  // Given an output tensor, find the input Variable with which it shares storage
+  auto get_shared_base = [&shared_pairs, &t2var](PyObject* tensor) -> Variable {
+    auto input_it = t2var.find(tensor);
+    if (input_it != t2var.end()) {
+      // If the output is an input, it's a view on itself
```

Comment:
The fact that it's in `t2var` inside this function doesn't mean it has to be an input - it can be one of the previous outputs. Not sure if this changes the logic